app/controllers/users_controller.rb

```diff
@@ -26,11 +26,7 @@ def show
 			@calories = self.class.calc_calories([@gender, @weight, @height, @age, @exercise, @goal]).round(0)
 			@all_recipes = Recipe.find_in_api(@calories, @time, @dietary_preferences)
 			if @all_recipes.nil? || @all_recipes.has_key?("hits")
-				if @all_recipes.nil?
-					flash[:alert] = "API limit reached, please try again in one minute!"
-				else
-					flash[:alert] = "We're sorry, the API did not return recipes for the data you specified"
-				end
+				flash[:alert] = set_alert
```

Comment:
ah, the code complexity was high because of these nested if clauses. good catch.